Sources/Sentry/SentryNSProcessInfoWrapper.mm

```diff
@@ -1,6 +1,26 @@
 #import "SentryNSProcessInfoWrapper.h"
 
-@implementation SentryNSProcessInfoWrapper
+@implementation SentryNSProcessInfoWrapper {
+#if TEST
+    NSString *_executablePath;
+}
+- (void)setProcessPath:(NSString *)path
+{
+    _executablePath = path;
+}
+#    define EXECUTABLE_PATH _executablePath;
+
+- (instancetype)init
+{
+    self = [super init];
+    _executablePath = NSBundle.mainBundle.bundlePath;
+    return self;
+}
+
+#else
+}
+#    define EXECUTABLE_PATH NSBundle.mainBundle.executablePath;
+#endif
```

Comment:
We should just avoid breaking up syntactically complete constructs with `#if`/`#else`. We only recently had a discussion about how complex it makes things here: https://github.com/getsentry/sentry-cocoa/pull/2858#discussion_r1159415510